frame/collective/src/lib.rs

```diff
@@ -152,19 +177,21 @@ decl_module! {
 
 		fn deposit_event() = default;
 
-		/// Set the collective's membership manually to `new_members`. Be nice to the chain and
-		/// provide it pre-sorted.
+		/// Set the collective's membership.
+		///
+		/// - `new_members`: The new member list. Be nice to the chain and
+		//	provide it sorted.
+		/// - `prime`: The prime member whose vote sets the default.
 		///
 		/// Requires root origin.
 		#[weight = SimpleDispatchInfo::FixedOperational(100_000)]
-		fn set_members(origin, new_members: Vec<T::AccountId>) {
+		fn set_members(origin, new_members: Vec<T::AccountId>, prime: Option<T::AccountId>) {
 			ensure_root(origin)?;
 			let mut new_members = new_members;
 			new_members.sort();
-			<Members<T, I>>::mutate(|m| {
-				<Self as ChangeMembers<T::AccountId>>::set_members_sorted(&new_members[..], m);
-				*m = new_members;
-			});
+			let old = Members::<T, I>::get();
+			<Self as ChangeMembers<T::AccountId>>::set_members_sorted(&new_members[..], &old);
+			Prime::<T, I>::set(prime);
```

Comment:
Updating `Members` storage with `new_members` has been lost with this change.

Reply:
yeah. that was superfluous logic as it happens anyway in `<Self as ChangeMembers<T::AccountId>>::set_members_sorted(&new_members[..], &old)`.

Reply:
I think this is an issue since other parts of logic rely on `Self::members()` which reads from the local members list.

We need some trait to tell the module where to read the correct list of members from, and use it here.

Reply:
Not sure what you mean. `Members` is always kept up to date regardless of how the set is changed.